lib/webrtc/rtcpeerconnection/chrome.js

```diff
@@ -24,351 +22,347 @@ var isUnifiedPlan = sdpUtils.getSdpFormat() === 'unified';
-    : {}, configuration);
-
-  util.interceptEvent(this, 'datachannel');
-  util.interceptEvent(this, 'signalingstatechange');
-
-  var sdpFormat = sdpUtils.getSdpFormat(newConfiguration.sdpSemantics);
-  var peerConnection = new RTCPeerConnection(newConfiguration, constraints);
-
-  Object.defineProperties(this, {
-    _appliedTracksToSSRCs: {
-      value: new Map(),
-      writable: true
-    },
-    _localStream: {
-      value: new MediaStream()
-    },
-    _peerConnection: {
-      value: peerConnection
-    },
-    _pendingLocalOffer: {
-      value: null,
-      writable: true
-    },
-    _pendingRemoteOffer: {
-      value: null,
-      writable: true
-    },
-    _rolledBackTracksToSSRCs: {
-      value: new Map(),
-      writable: true
-    },
-    _sdpFormat: {
-      value: sdpFormat
-    },
-    _senders: {
-      value: new Map()
-    },
-    _signalingStateLatch: {
-      value: new Latch()
-    },
-    _tracksToSSRCs: {
-      value: new Map(),
-      writable: true
-    },
-    localDescription: {
-      enumerable: true,
-      get: function() {
-        return this._pendingLocalOffer ? this._pendingLocalOffer : peerConnection.localDescription;
-      }
-    },
-    remoteDescription: {
-      enumerable: true,
-      get: function() {
-        return this._pendingRemoteOffer ? this._pendingRemoteOffer : peerConnection.remoteDescription;
-      }
-    },
-    signalingState: {
-      enumerable: true,
-      get: function() {
-        if (this._pendingLocalOffer) {
-          return 'have-local-offer';
-        } else if (this._pendingRemoteOffer) {
-          return 'have-remote-offer';
+class ChromeRTCPeerConnection extends EventTarget {
+  constructor(configuration = {}, constraints) {
+    super();
+    const newConfiguration = Object.assign(configuration.iceTransportPolicy
+      ? { iceTransports: configuration.iceTransportPolicy }
+      : {}, configuration);
+
+    interceptEvent(this, 'datachannel');
+    interceptEvent(this, 'signalingstatechange');
+    const sdpFormat = getSdpFormat(newConfiguration.sdpSemantics);
+    const peerConnection = new RTCPeerConnection(newConfiguration, constraints);
+
+    Object.defineProperties(this, {
+      _appliedTracksToSSRCs: {
+        value: new Map(),
+        writable: true
+      },
+      _localStream: {
+        value: new MediaStream()
+      },
+      _peerConnection: {
+        value: peerConnection
+      },
+      _pendingLocalOffer: {
+        value: null,
+        writable: true
+      },
+      _pendingRemoteOffer: {
+        value: null,
+        writable: true
+      },
+      _rolledBackTracksToSSRCs: {
+        value: new Map(),
+        writable: true
+      },
+      _sdpFormat: {
+        value: sdpFormat
+      },
+      _senders: {
+        value: new Map()
+      },
+      _signalingStateLatch: {
+        value: new Latch()
+      },
+      _tracksToSSRCs: {
+        value: new Map(),
+        writable: true
+      },
+      localDescription: {
```

Comment:
Using 'get' to declare public properties outside the constructor seemed to be causing errors where some properties were missing when the object was being accessed/printed out.